apps/emqx/src/emqx_channel.erl

```diff
@@ -1639,14 +1639,18 @@ handle_info({disconnect, ReasonCode, ReasonName, Props}, Channel) ->
     handle_out(disconnect, {ReasonCode, ReasonName, Props}, Channel);
 handle_info({puback, PacketId, PubRes, RC}, Channel) ->
     do_finish_publish(PacketId, PubRes, RC, Channel);
-handle_info({'DOWN', Ref, process, Pid, Reason}, Channel) ->
-    case emqx_hooks:run_fold('client.monitored_process_down', [Ref, Pid, Reason], []) of
-        [] -> {ok, Channel};
-        Msgs -> {ok, Msgs, Channel}
-    end;
-handle_info(Info, Channel = #channel{session = Session0, clientinfo = ClientInfo}) ->
+handle_info(Info, Channel0 = #channel{session = Session0, clientinfo = ClientInfo}) ->
```

Comment:
When subscription dies on the DS side, the session gets a DOWN message. Therefore now we should pass DOWN messages to the session.